### PR TITLE
add: Redis CCR feature

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes

Content for the Redis cross-cluster replication (CCR) feature as part of the Pro Platform.
[DOC-853](https://aiven.atlassian.net/browse/DOC-853)

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

[DOC-853]: https://aiven.atlassian.net/browse/DOC-853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/redis/concepts/redis-ccr.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ensuring high availability and data resilience across deployments.
+
+**Replication process:** The replication process involves two services - the primary and
+replica services. The system automatically and asynchronously copies any new data you
+add to the primary service to the replica service. The primary service manages all
+read and write operations for optimal performance and data consistency under
+normal conditions.
+
+**Handling failures:** In case of any failure of the primary service, Aiven provides
+automatic failover functionality, which promotes the replica service to become
```
Comment (bogdanp05) [inline docs/products/redis/concepts/redis-ccr.md:33]: it's not automatic afaik. I never implemented this feature and I don't know if it was a pre-existing feature of the integration type. 
Manual promotion of the replica service does work, though.

File: docs/products/redis/concepts/redis-ccr.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+the new primary, maintaining service continuity.
+
+However, to fully resume normal operations, you must manually update your application’s
+connection settings to point to the new primary service.
+
+## Limitations
+
+- This feature requires `[Pro Platform](/docs/platform/concepts/pro-platform)`.
+- Aiven for Redis CCR follows an active-passive model, which means that only the data
```
Comment (bogdanp05) [inline docs/products/redis/concepts/redis-ccr.md:42]: it's actually not possible to write data in the replica service at all while the integration is active.

File: docs/products/redis/concepts/redis-ccr.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ingested in the primary service gets replicated.
+- Aiven for Redis services only support replication within their own service type.
+Cross-service replication to or from Aiven for Dragonfly services is not supported.
+- Enabling up CCR for Aiven for Redis is only possible via API/CLI, with console support
```
Comment (bogdanp05) [inline docs/products/redis/concepts/redis-ccr.md:46]: Console support is a one day job. I actually implemented it myself just for demo purposes. I assumed when we launch this, we would also have console support, or?

File: docs/products/redis/howto/enable-redis-ccr.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+   your primary Aiven for Redis service has a backup.
+   [Configuring backups](/docs/platform/concepts/service_backups#aiven-for-redis)
+   is essential to ensure data consistency and integrity for replication.
+
+1. **Create the replica service:** Next, create the Aiven for Redis service that acts as
+   the replica. Specify the primary service as the source for replication using the
+   following API call:
+
+   ```shell
+    curl -X POST https://api.aiven.io/v1/project/PROJECT_NAME/service \
```
Comment (bogdanp05) [inline docs/products/redis/howto/enable-redis-ccr.md:56]: commands looks good but didn't run them myself now. I assume you did and they work?

File: docs/products/redis/howto/enable-redis-ccr.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+   - `PRIMARY_SERVICE_NAME`: The name of your primary service.
+
+## Verify data replication
+
+After setting up the replica, ensure that data entered into the primary Aiven for Redis
+service is successfully replicated to the replica service.
+
+## Manage failover
+
+If the primary service fails, Aiven automatically promotes the replica service to the
```
Comment (bogdanp05) [inline docs/products/redis/howto/enable-redis-ccr.md:127]: same point as above about automatic failover. To manually promote the replica to master, you'd have to delete the integration.

**Review by bogdanp05 — CHANGES_REQUESTED** [review]
this should only be merged once we actually enable Redis CCR for pro-platform users, right? Which we haven't done yet.

File: docs/products/redis/howto/enable-redis-ccr.md (hunk 4 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/howto/enable-redis-ccr.md:56]: I didn't either; I based it on the examples in other services (PG, for example).

File: docs/products/redis/concepts/redis-ccr.md (hunk 3 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/redis-ccr.md:46]: If the feature support is available on the Aiven console at the time of release, I can include it in the documentation. However, as per the information provided by @hoang-vo, the console updates were expected to be released in the next quarter or so.

File: docs/products/redis/concepts/redis-ccr.md (hunk 1 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/redis-ccr.md:33]: If the failover process is manual, could you provide more details and steps on how users can promote a replica to primary during a primary cluster failure?

File: docs/products/redis/concepts/redis-ccr.md (hunk 2 above)
Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/redis-ccr.md:42]: Just to confirm, when you say “it’s not possible to write data in the replica service during active integration”, are you suggesting that in Aiven for Redis CCR’s active-passive model, all new data is added to the primary service first and then replicated to the replica service? Could you please confirm or provide more details if necessary?

Comment (harshini-rangaswamy) [reply docs/products/redis/concepts/redis-ccr.md:42]: how about rewording it to this: 

Aiven for Redis CCR uses an active-passive model where data can be written only to the primary service and then replicated to the replica service.

Comment (bogdanp05) [reply docs/products/redis/concepts/redis-ccr.md:42]: Yes. And the rewrite looks good.

File: docs/products/redis/concepts/redis-ccr.md (hunk 3 above)
Comment (bogdanp05) [reply docs/products/redis/concepts/redis-ccr.md:46]: Oh ok, we can confirm with Hoang once he's back.

**Comment by harshini-rangaswamy** [comment]
Service is reaching EOA and EOL soon.
